pkg/ipam/node.go

```diff
@@ -219,34 +219,28 @@ func (n *Node) loggerLocked() (logger *logrus.Entry) {
 func (n *Node) getMaxAboveWatermark() int {
 	if n.resource.Spec.IPAM.MaxAboveWatermark != 0 {
 		return n.resource.Spec.IPAM.MaxAboveWatermark
-	}
-	// OBSOLETE: This can be removed in Cilium 1.9
-	return n.resource.Spec.ENI.MaxAboveWatermark
+	}	
+	return nil
```

Comment:
This does not build. I think you can just return the value here, no need for the if anymore statement.

Reply:
Thank you for your comments. Fixed now :)